### DISCUSSION TITLE
how can i install Flask manually without using "pip install -U Flask"?

### DISCUSSION THREAD

**Discussion by yyhk123** [pallets/flask#5292] [Q&A]
Hello,
I am trying to install Flask package manually but there is no setup.py inside the folder (where i got the file from https://pypi.org/project/Flask/)

i am trying to install package on the server but server does not take pip install flask command so i have been installing packages manually with setup.py but this is missing in the file.

How can i install it manually??

Thank you!

**Comment by ThiefMaster** [comment]
Fix your environment: Install pip. If there's no network available, you can still download the packages from PyPI and install those using pip.

**Comment by umair313** [comment]
try with poetry if the pip is not working using a Package manager has many advantages over doing it manually.
